azure/durable_functions/orchestrator.py

```diff
@@ -135,12 +141,17 @@ def _add_to_actions(self, generation_state):
         if self.durable_context.will_continue_as_new:
             return
         if not generation_state._is_yielded:
-            if (isinstance(generation_state, Task)
-                    and hasattr(generation_state, "action")):
-                self.durable_context.actions.append([generation_state.action])
-            elif (isinstance(generation_state, TaskSet)
-                    and hasattr(generation_state, "actions")):
-                self.durable_context.actions.append(generation_state.actions)
+            if isinstance(generation_state, Task):
+                if self.durable_context._replay_schema == ReplaySchema.V1:
+                    self.durable_context.actions.append([generation_state.action])
+                else:
+                    self.durable_context.actions[0].append(generation_state.action)
```

Comment:
Just to double check, V2 schema expects 
`[[action1, action2, action3]]` and not `[[action1], [action2], [action3]]`? right?

Reply:
That's correct! I wish V2 would simply expect a flat list, instead of a doubly-nested one, but there were backwards compatibility reasons that made that difficult

In the future, we'll should make changes in the C# layer to remove the need for a nested list of actions